src/features/governance/utils/events/vote.ts

```diff
@@ -0,0 +1,41 @@
+import 'server-only';
+
+import { Event } from 'src/app/governance/events';
+import { votesTable } from 'src/db/schema';
+import { VoteType } from 'src/features/governance/types';
+import {
+  assertEvent,
+  decodeVoteEventLog,
+  fetchProposalVoters,
+} from 'src/features/governance/utils/votes';
+
+const ALLOWED_EVENTS = [
+  'ProposalVoted',
+  'ProposalVotedV2',
+  'ProposalVoteRevoked',
+  'ProposalVoteRevokedV2',
+] as const;
+export async function handleVoteEvent(
+  eventName: string,
+  event: Event,
+  chainId: number,
+): Promise<(typeof votesTable)['$inferInsert'][]> {
+  if (!assertEvent(ALLOWED_EVENTS, eventName)) {
+    console.info('Not a vote event');
+    return [];
+  }
+
+  const proposal = decodeVoteEventLog(event);
+  if (!proposal || !proposal.proposalId) {
+    throw new Error('Couldnt decode the vote event: ' + JSON.stringify(event));
+  }
+
+  const { totals } = await fetchProposalVoters(proposal.proposalId);
```

Comment:
this is such a nit but.. fetch* sounds like we are remotely fetching from a server over http but this is more like `sumProposalVotes`